clients/redigo.go

```diff
@@ -0,0 +1,389 @@
+//
+func (r *Redigo) JSONArrAppend(key, path string, values ...interface{}) (res interface{}, err error) {
+
+	if len(values) == 0 {
+		return nil, rjs.NeedAtleastOneArg
+	}
+	args := make([]interface{}, 0)
+	args = append(args, key, path)
+	args = append(args, values...)
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_ARRAPPEND, args...)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONArrLen returns the length of the json array at path
+//
+// ReJSON syntax:
+// 	JSON.ARRLEN <key> [path]
+//
+func (r *Redigo) JSONArrLen(key, path string) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_ARRLEN, key, path)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONArrPop removes and returns element from the index in the array
+// to pop last element use rejson.PopArrLast
+//
+// ReJSON syntax:
+// 	JSON.ARRPOP <key> [path [index]]
+//
+func (r *Redigo) JSONArrPop(key, path string, index int) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_ARRPOP, key, path, index)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONArrIndex returns the index of the json element provided and return -1 if element is not present
+//
+// ReJSON syntax:
+// 	JSON.ARRINDEX <key> <path> <json-scalar> [start [stop]]
+//
+func (r *Redigo) JSONArrIndex(key, path string, jsonValue interface{}, optionalRange ...int) (
+	res interface{}, err error) {
+
+	args := []interface{}{key, path, jsonValue}
+
+	ln := len(optionalRange)
+	switch {
+	case ln > 2:
+		return nil, rjs.NeedAtleastOneArg
+	case ln == 1: // only inclusive start is present
+		args = append(args, optionalRange[0])
+	case ln == 2: // both inclusive start and exclusive end are present
+		args = append(args, optionalRange[0], optionalRange[1])
+	}
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_ARRINDEX, args...)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONArrTrim trims an array so that it contains only the specified inclusive range of elements
+//
+// ReJSON syntax:
+// 	JSON.ARRTRIM <key> <path> <start> <stop>
+//
+func (r *Redigo) JSONArrTrim(key, path string, start, end int) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_ARRTRIM, key, path, start, end)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONArrInsert inserts the json value(s) into the array at path before the index (shifts to the right).
+//
+// ReJSON syntax:
+// 	JSON.ARRINSERT <key> <path> <index> <json> [json ...]
+//
+func (r *Redigo) JSONArrInsert(key, path string, index int, values ...interface{}) (res interface{}, err error) {
+
+	if len(values) == 0 {
+		return nil, rjs.NeedAtleastOneArg
+	}
+	args := make([]interface{}, 0)
+	args = append(args, key, path, index)
+	args = append(args, values...)
```

Comment:
Combine the appends

Reply:
https://github.com/nitishm/go-rejson/pull/29#discussion_r250390783